main.js

```diff
@@ -650,10 +685,20 @@
                 return soFar.then(f);
             }, Q.call());
         })
+        .then(function () {
+            if (!!_config["results-xml-path"]) {
+                var resultsXmlPath = _config["results-xml-path"];
+                _logger.info("Writing XML results to:  " , resultsXmlPath);
+                return Q.ninvoke(fse,"writeFile", resultsXmlPath, xmlResults(results));
+            } else {
+                return new Q(true) ;
```

Comment:
seems like maybe this whole `else` wasn't necessary?  Or could have just returned `true`?  not that this matters much

Reply:
This is the bit I was talking about with you earlier.  I will be cleaning up this section, perhaps using the .tap method.